NORTRIP_multiroad_read_metcoop_netcdf4.f90

```diff
@@ -277,131 +277,179 @@ subroutine NORTRIP_read_metcoop_netcdf4
 
     !Convert dew point to RH if RH not available and dewpoint is
     if (var_available_nc(dewpoint_index).and..not.var_available_nc(relhumidity_index)) then
         do k=1,size(var3d_nc,4)
-        do j=1,size(var3d_nc,3)
-        do i=1,size(var3d_nc,2)
-            var3d_nc(relhumidity_index,i,j,k)=RH_from_dewpoint_func(var3d_nc(temperature_index,i,j,k)-TOC,var3d_nc(dewpoint_index,i,j,k)-TOC)/100.
-            var3d_nc(relhumidity_index,i,j,k)=max(var3d_nc(relhumidity_index,i,j,k),0.)
-            var3d_nc(relhumidity_index,i,j,k)=min(var3d_nc(relhumidity_index,i,j,k),1.)
-        enddo
-        enddo
+            do j=1,size(var3d_nc,3)
+                do i=1,size(var3d_nc,2)
+                    var3d_nc(relhumidity_index,i,j,k)=RH_from_dewpoint_func(var3d_nc(temperature_index,i,j,k)-TOC,var3d_nc(dewpoint_index,i,j,k)-TOC)/100.
+                    var3d_nc(relhumidity_index,i,j,k)=max(var3d_nc(relhumidity_index,i,j,k),0.)
+                    var3d_nc(relhumidity_index,i,j,k)=min(var3d_nc(relhumidity_index,i,j,k),1.)
+                enddo
+            enddo
         enddo
     endif
     
     !Calculate angle difference between North and the Model Y direction based on the middle grids
-    !Not correct, needs to be fixed
+    !Not correct, needs to be fixed !TODO: Is this fixed?
     i_grid_mid=int(dim_length_nc(x_index)/2)
     j_grid_mid=int(dim_length_nc(y_index)/2)
-    dgrid_nc(x_index)=var1d_nc(x_index,i_grid_mid)-var1d_nc(x_index,i_grid_mid-1)
-    dgrid_nc(y_index)=var1d_nc(y_index,j_grid_mid)-var1d_nc(y_index,j_grid_mid-1)
+    dgrid_nc(x_index)=var1d_nc_old(x_index,i_grid_mid)-var1d_nc_old(x_index,i_grid_mid-1)
+    dgrid_nc(y_index)=var1d_nc_old(y_index,j_grid_mid)-var1d_nc_old(y_index,j_grid_mid-1)
     dlat_nc=var2d_nc(lat_index,i_grid_mid,j_grid_mid)-var2d_nc(lat_index,i_grid_mid,j_grid_mid-1)
     
     !If the coordinates are in km instead of metres then change to metres (assuming the difference is not going to be > 100 km
     if (dgrid_nc(x_index).lt.100) then
         dgrid_nc=dgrid_nc*1000.
-        var1d_nc(x_index,:)=var1d_nc(x_index,:)*1000.
-        var1d_nc(y_index,:)=var1d_nc(y_index,:)*1000.
+        var1d_nc_old(x_index,:)=var1d_nc_old(x_index,:)*1000.
+        var1d_nc_old(y_index,:)=var1d_nc_old(y_index,:)*1000.
     endif
     
     angle_nc=180./3.14159*acos(dlat_nc*3.14159/180.*6.37e6/dgrid_nc(x_index))
     write(unit_logfile,'(A,2f12.1)') ' Grid spacing X and Y (m): ', dgrid_nc(x_index),dgrid_nc(y_index)
     write(unit_logfile,'(A,2i,f12.4)') ' Angle difference between grid and geo North (i,j,deg): ', i_grid_mid,j_grid_mid,angle_nc
 
+
+
+    meteo_nc_timesteps = nint(1 + (dim_length_nc(time_index)-1)/timestep) !Number of time steps that will be saved from the meteo file. (If timestep = 1h this will just be the number of hours)
+
+    !Fill a date_nc array that is used to match meteo dates to the date range specified in the simulation call. 
+    allocate(date_nc(num_date_index,meteo_nc_timesteps))
```

Comment:
I have moved the definition of date_nc from NORTRIP_multiroad_save_meteodata.f90 to here. I'm sure I had a good reason for doing that related to the shorter timesteps, but I cannot right now remember why. Could it cause problems, or is it ok to leave it here?